owncloudApp/src/main/java/com/owncloud/android/shares/ui/fragment/PublicShareDialogFragment.kt

```diff
@@ -291,30 +313,57 @@ class PublicShareDialogFragment : DialogFragment() {
         // since the public link permission foo got a bit despagetified in the server somewhere
         // at 10.0.4 we don't need publicUploadPermission there anymore. By setting it to false
         // it will not be sent to the server.
-
-        publicUploadPermission =
-            mCapabilities!!.versionMayor >= 10 && (mCapabilities!!.versionMinor > 1 || mCapabilities!!.versionMicro > 3) && publicUploadPermission
+        if (capabilities != null) {
+            publicUploadPermission =
+                capabilities!!.versionMayor >= 10 &&
```

Comment:
can we move those hardcoded values to constants??

Reply:
done